toucan_connectors/mongo/mongo_connector.py

```diff
@@ -260,15 +261,25 @@ def _execute_query(self, data_source: MongoDataSource):
             col = client[data_source.database][data_source.collection]
             return col.aggregate(data_source.query)  # type: ignore[arg-type]
 
-    def _retrieve_data(self, data_source):
+    def _retrieve_data(self, data_source, chunk_size: int | None = None):
         data_source.query = normalize_query(data_source.query, data_source.parameters)
         data = self._execute_query(data_source)
-        return pd.DataFrame(list(data))
+
+        if chunk_size:
+            chunks = []
+            while True:
+                chunk = list(itertools.islice(data, chunk_size))
+                if not chunk:
+                    break
```

Comment:
```suggestion
            while (chunk := list(itertools.islice(data, chunk_size))):
```

On s'évite le `while True` comme ça